src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function updateTransferSettlement (transferId, status, trx = null) {
+  Logger.info(Utility.breadcrumb(location, { method: 'updateTransferSettlement' }))
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        // Insert TransferParticipant ledger entry type.
+        await knex.from(knex.raw('transferParticipant (transferID, participantCurrencyId, transferParticipantRoleTypeId, ledgerEntryTypeId, amount)'))
+          .insert(function () {
+            this.from('transferParticipant AS TP')
+              .select('TP.transferId', 'TP.participantCurrencyId', 'TP.transferParticipantRoleTypeId', 'TP.ledgerEntryTypeId', knex.raw('?? * -1', ['TP.amount']))
+              .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+              .innerJoin('settlementModel AS M', 'PC.ledgerAccountTypeId', 'M.ledgerAccountTypeId')
+              .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+              .where(function () {
+                this.where({ 'TP.transferId': transferId })
+                this.andWhere(function () {
+                  this.andWhere({ 'G.name': 'GROSS' })
+                })
+              })
+              .union(function () {
+                this.select('TP.transferId', 'PC1.participantCurrencyId', 'TP.transferParticipantRoleTypeId', 'TP.ledgerEntryTypeId', 'TP.amount')
+                  .from('transferParticipant AS TP')
+                  .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+                  .innerJoin('settlementModel AS M', 'PC.ledgerAccountTypeId', 'M.ledgerAccountTypeId')
+                  .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+                  .innerJoin('participantCurrency AS PC1', function () {
+                    this.on('PC1.currencyId', 'PC.currencyId')
+                      .andOn('PC1.participantId', 'PC.participantId')
+                      .andOn('PC1.ledgerAccountTypeId', 'M.settlementAccountTypeId')
+                  })
+                  .where(function () {
+                    this.where({ 'TP.transferId': transferId })
+                    this.andWhere(function () {
+                      this.andWhere({ 'G.name': 'GROSS' })
+                    })
+                  })
+              })
+          })
+          .transacting(trx)
+
+        // Insert a new status for the transfer.
+        const transferStateChange = [
+          {
+            transferId: transferId,
+            transferStateId: 'SETTLED',
```

Comment:
transferStateId should be set using an ENUM from central-services-shared.

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story If appropriate ENUMS are not already available for this.